backend_new/test/unit/services/application.service.spec.ts

```diff
@@ -0,0 +1,516 @@
+    prisma.applications.findFirst = jest.fn().mockResolvedValue({
+      id: 'example id',
+    });
+
+    const params: ApplicationQueryParams = {
+      orderBy: ApplicationOrderByKeys.createdAt,
+      order: OrderByEnum.ASC,
+      listingId: 'example listing id',
+      limit: 3,
+      page: 1,
+    };
+
+    expect(await service.list(params)).toEqual({
+      items: [
+        {
+          id: `application id 0`,
+          appUrl: `appUrl 0`,
+          additionalPhone: true,
+          additionalPhoneNumber: `additionalPhoneNumber 0`,
+          additionalPhoneNumberType: `additionalPhoneNumberType 0`,
+          householdSize: 0,
+          housingStatus: `housingStatus 0`,
+          sendMailToMailingAddress: true,
+          householdExpectingChanges: true,
+          householdStudent: true,
+          incomeVouchers: true,
+          income: `income 0`,
+          incomePeriod: IncomePeriodEnum.perMonth,
+          preferences: {
+            claimed: undefined,
+            key: undefined,
+            options: undefined,
+          },
+          status: ApplicationStatusEnum.submitted,
+          submissionType: ApplicationSubmissionTypeEnum.electronical,
+          acceptedTerms: true,
+          submissionDate: date,
+          markedAsDuplicate: false,
+          confirmationCode: `confirmationCode 0`,
+          reviewStatus: ApplicationReviewStatusEnum.valid,
+          applicant: {
+            firstName: `application 0 firstName`,
+            middleName: `application 0 middleName`,
+            lastName: `application 0 lastName`,
+            birthMonth: `application 0 birthMonth`,
+            birthDay: `application 0 birthDay`,
+            birthYear: `application 0 birthYear`,
+            emailAddress: `application 0 emailaddress`,
+            noEmail: false,
+            phoneNumber: `application 0 phoneNumber`,
+            phoneNumberType: `application 0 phoneNumberType`,
+            noPhone: false,
+            workInRegion: YesNoEnum.yes,
+            applicantWorkAddress: {
+              placeName: `application {i} applicantWorkAddress placeName`,
+              city: `application {i} applicantWorkAddress city`,
+              county: `application {i} applicantWorkAddress county`,
+              state: `application {i} applicantWorkAddress state`,
+              street: `application {i} applicantWorkAddress street`,
+              street2: `application {i} applicantWorkAddress street2`,
+              zipCode: `application {i} applicantWorkAddress zipCode`,
+              latitude: 0,
+              longitude: 0,
+            },
+            applicantAddress: {
+              placeName: `application {i} applicantAddress placeName`,
+              city: `application {i} applicantAddress city`,
+              county: `application {i} applicantAddress county`,
+              state: `application {i} applicantAddress state`,
+              street: `application {i} applicantAddress street`,
```

Comment:
suggestion: setup some more variables for better reusability for tests

Reply:
updated